File: Level2/카펫.py
brown = 24
yellow = 24

def solution(brown, yellow):
    answer = []
    lis = []
    for i in range(1, brown + yellow + 1):
        if (brown + yellow) % i == 0:
            lis.append(i)

    for i in lis:
        for k in lis:
            if i * k == brown + yellow:
                if (i-2)*(k-2) == yellow:
                    answer.append(i)
                    answer.append(k)
                    answer.sort()
                    answer.reverse()
                    return answer

did = solution(brown, yellow)
print(did)

File: Level2/카펫.java
import java.util.*;

class Solution {
    public int[] solution(int brown, int yellow) {
        int size = brown + yellow;
        int[] answer = new int[2];
        for(int i=1; i<=size; i++) {
            if(size % i > 0)
                continue;
            else {
                int col = (size / i);
                if(((i*2) + ((col-2)*2) == brown) && ((i-2)*(col-2) == yellow)) {
                    answer[0] = Math.max(i, col);
                    answer[1] = Math.min(i, col);
                    break;
                }
            }
        }
        return answer;
    }
}
